### ISSUE TITLE
Leaving a challenge and removing  tasks does not remove challenge's tags

### ISSUE THREAD

**Issue by CuriousMagpie** [HabitRPG/habitica#13844]
### General Info
  * UUID: 09d36e6e-229e-4d50-85cf-ed870098174c
  * Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36

### Description
From user:

> When I Leave a Challenge and hit Remove Tasks, It does not also remove that challenge's tag from either Tags List in Tasks tab

I've noticed this as well and it's vaguely irritating. I think it would be a good upgrade to change `website/client/src/components/challenges/leaveChallengeModal.vue` so that when it removes the tasks it also cleans up the challenge's `shortName` as well.

**Comment by SabreCat** [comment]
I believe this is intentional. The tag becomes part of the user's list, and they can apply it to non-Challenge tasks if desired.

So if we do implement a change here, we'd at least want to check that the user hasn't made use of the tag elsewhere before deleting it.

**Comment by megansearles** [comment]
Hi! Can I take this issue?

**Comment by CuriousMagpie** [comment]
@megansearles Absolutely--just keep in mind there needs to be a check that the user hasn't used the tag elsewhere!
